src/main/java/seedu/address/model/person/StudentId.java

```diff
@@ -0,0 +1,53 @@
+package seedu.address.model.person;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Student's matriculation number in TAPA.
+ * Guarantees: immutable; is valid as declared in {@link #isValidId(String)}
+ */
+public class StudentId {
+    public static final String MESSAGE_CONSTRAINTS = "Matriculation number should only contain alphanumeric characters";
+    public static final String VALIDATION_REGEX = "[a-zA-Z0-9]*";
+
+    public final String id;
```

Comment:
Do yall think we should change this to private field and use getters/setters instead?

Reply:
After looking through the other classes related to person (eg. name, address etc.), you're right that they all use `public final`. TBH I'm not too sure why but if I had to guess, it might have something to do with Junit testing. In iP, I switched some fields to `private` and certain Junit tests stopped working.

I think what you have now (using `public final`) should be fine since all other classes always call the constructors to make a new instance of (for example) the name object whenever they want to change the name.